main.cpp

```diff
@@ -0,0 +1,92 @@
+#include <iostream>
+#include <vector>
+
+using namespace std;
+
+class rs_m
+{
+public:
+    vector<double> r; 

```

Comment:
The variable name is hard to understand, what does this vector hold? By my best guess it's room numbers, and If I'm right, I'd suggest to name this variable "roomNumbers"?

Also, if this variable is actually for holding room numbers, maybe use `int` to save memory?